src/Microsoft.TemplateEngine.Edge/Template/ParameterSetBuilder.cs

```diff
@@ -0,0 +1,361 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Text;
+using Microsoft.Extensions.Logging;
+using Microsoft.TemplateEngine.Abstractions;
+using Microsoft.TemplateEngine.Abstractions.Parameters;
+using Microsoft.TemplateEngine.Utils;
+
+namespace Microsoft.TemplateEngine.Edge.Template
+{
+#pragma warning disable CS0618 // Type or member is obsolete
+    internal class ParameterSetBuilder : ParametersDefinition, IParameterSetBuilder, IParameterSet
```

Comment:
nit: it would be good to implement obsolete interface explicitly, likely in different file. This should allow not to mark the type as obsolete (but not sure).

Reply:
The class itself is not obsolete. Though there is space to make it more clean.

I moved the pragma so that it applies only to the deprecated interface, implemented the interface explicitly and moved all members to single region.